docs/design-documents/promtail-scraping-objectstore.md

```diff
@@ -0,0 +1,89 @@
+
+### How should we support reading of compressed files ?
+
+If the files in the Objectstore is stored in a compressed format, promtail should have support for reading the compressed data and we should make sure we support most of the compression algorithms like we already do for chunks(Snappy, LZ4, Gzip) 
+
+### How we should identify and watch newly added files ?
+
+To identify and to start tailing the newly added files, we should list the files in the Object store at some desired intervals. How frequently we should this? Once we add the file to `watch`, to identify any changes to the file we should frequently check the modified `datetime` of the file. The list of files might be large and each watch will be running in a `go routine` and each `go routine` making frequent `http` requests to the Obejctstore. We should make sure this doesn't affect promtail's performance.
+
+## Proposed Solution for the above mentioned points
+
+> There is no appending of lines to file in object store
+
+In the positions.yaml along with the `offset` we should also store file modified `datetime`. If we see that date is modified we should reset the offset and tail the contents from the begining. This is the same behaviour in local filesystem as well. If a file is `opened`, `hpcloud tail` will tail the file from the begining.
+
+#### Alternative 
+
+Instead of tailing the file from the beging upon modification, we can calculate the `checksum` of the contents till the `offset`, and check if the checksum was changed or not after the file was modified. If it is not changed then we can just proceed further, else we can tail from the begining.
+
+Any other better way to handle this?
+
+
+> Where should we keep `positions.yaml` file ?
+
+I think this should be an user choice. One thing we need to consider here is that there will be API limit per account per region and if we store the `positions.yaml` in the object store itself, we may end up doing lot of APIs calls and we would get `RequestLimitExceeded` very frequently. Still, backoff and retry should be implemented in case we get this error even when doing API calls less frequently. So, placing and updating the `positions.yaml` should be left to the user so he can decide based on the API usage of his setup wether to keep it locally or remotely. 
+
+> How we should identify and watch newly added files ?
+
+Limit on API calls should also be considered for syncing the newly added files. To start tailing the newly added files, we should do `ListObjects` at a desired interval. We should make sure we don't keep this interval too small.
+Also, once the entire contents of the file is read we should stop watching the file so that we reduce the number of active `go routines`. 
+
+> How should we support reading of compressed files ?
+
+It would be good to support most of the compression algorithms like we already do for chunks(Snappy, LZ4, Gzip). 
```

Comment:
Let's add this after, it will be super easy.

Reply:
Sure